url/s3url.go

```diff
@@ -24,38 +40,112 @@ func (s S3Url) Format() string {
 	if s.Key == "" {
 		return s.Bucket
 	}
-	return s.Bucket + "/" + s.Key
+	return s.Bucket + s3Separator + s.Key
+}
+
+// setPrefixAndFilter sets prefix and filter parts on s3 key
+//
+// filter is the part that comes after the wildcard string
+// prefix is the part that comes before the wildcard string
+func (s *S3Url) setPrefixAndFilter() {
+	loc := strings.IndexAny(s.Key, s3WildCharacters)
+	wildOperation := loc > -1
+	if !wildOperation {
+		s.Delimiter = s3Separator
+		s.Prefix = s.Key
+	} else {
+		s.Prefix = s.Key[:loc]
+		s.filter = s.Key[loc:]
+	}
+}
+
+// setFilterRegex converts wildcard strings to regex format
+// and precompiles it for later usage. It is default to
+// ".*" to match every key on S3
+func (s *S3Url) setFilterRegex() error {
```

Comment:
we should merge this func with `setPrefixAndFilter`. No need for a separate function.

Reply:
done